MuxExoPlayer/src/main/java/com/mux/stats/sdk/muxstats/MuxBaseExoPlayer.java

```diff
@@ -72,44 +80,40 @@
     protected Float sourceAdvertisedFramerate;
     protected Long sourceDuration;
     protected ExoPlayerHandler playerHandler;
+    protected FrameRenderedListener frameRenderedListener;
+    protected Timer updatePlayheadPositionTimer;
 
     protected WeakReference<ExoPlayer> player;
     protected WeakReference<View> playerView;
     protected WeakReference<Context> contextRef;
+    protected AdsImaSDKListener adsImaSdkListener;
 
     protected int streamType = -1;
 
     public enum PlayerState {
-        BUFFERING, ERROR, PAUSED, PLAY, PLAYING, PLAYING_ADS, FINISHED_PLAYING_ADS, INIT, ENDED
+        BUFFERING, REBUFFERING, SEEKING, SEEKED, ERROR, PAUSED, PLAY, PLAYING, PLAYING_ADS,
+        FINISHED_PLAYING_ADS, INIT, ENDED
     }
     protected PlayerState state;
     protected MuxStats muxStats;
 
 
     MuxBaseExoPlayer(Context ctx, ExoPlayer player, String playerName,
                      CustomerPlayerData customerPlayerData, CustomerVideoData customerVideoData,
-                     CustomerViewData customerViewData, boolean sentryEnabled) {
+                     CustomerViewData customerViewData, boolean sentryEnabled,
+                     INetworkRequest networkRequest) {
         super();
         this.player = new WeakReference<>(player);
         this.contextRef = new WeakReference<>(ctx);
         state = PlayerState.INIT;
         MuxStats.setHostDevice(new MuxDevice(ctx));
-        MuxStats.setHostNetworkApi(new MuxNetworkRequests());
+        MuxStats.setHostNetworkApi(networkRequest);
         muxStats = new MuxStats(this, playerName, customerPlayerData, customerVideoData, customerViewData, sentryEnabled);
         addListener(muxStats);
-        Player.VideoComponent lDecCount = player.getVideoComponent();
         playerHandler = new ExoPlayerHandler(player.getApplicationLooper(), player);
-        lDecCount.setVideoFrameMetadataListener(new VideoFrameMetadataListener() {
-            // As of r2.11.x, the signature for this callback has changed. These are not annotated as @Overrides in
-            // order to support both before r2.11.x and after r2.11.x at the same time.
-            public void onVideoFrameAboutToBeRendered(long presentationTimeUs, long releaseTimeNs, Format format) {
-                playerHandler.obtainMessage(ExoPlayerHandler.UPDATE_PLAYER_CURRENT_POSITION).sendToTarget();
-            }
-
-            public void onVideoFrameAboutToBeRendered(long presentationTimeUs, long releaseTimeNs, Format format, @Nullable MediaFormat mediaFormat) {
-                playerHandler.obtainMessage(ExoPlayerHandler.UPDATE_PLAYER_CURRENT_POSITION).sendToTarget();
-            }
-        });
+        frameRenderedListener = new FrameRenderedListener(playerHandler);
+        setPlaybackHeadUpdateInterval(false);
+        adsImaSdkListener = new AdsImaSDKListener(this);
```

Comment:
We can't do this here, we need to do it only within the call to `monitorImaAdsLoader`, so that we can move the dependency to `compileOnly`. Otherwise, we're going to bundle all of IMA within our SDK, which is not what we want to do, and introduce a hard dependency on that library.

Reply:
I tested, it can stay here, I changed the googled dependencies  to compileOnly.